### ISSUE TITLE
FAQ: updating certain packages while a kernel is running is going to break the installation/fail

### ISSUE THREAD

**Issue by jankatins** [IRkernel/irkernel.github.io#32]
If update.packages touches a native package which is loaded, it will fail and leave the package in a weired state (at least if I use `R CMD build` in my local git). 

This is not a problem unless that that package is jsonlite, which the kernel needs... In that case you need to shutdown all kernels and then reinstall jsonlite...

**Comment by takluyver** [comment]
Is this a Windows issue with multiple processes trying to open the same file?

**Comment by jankatins** [comment]
No, I think it is a problem with native code (= `*.so` on linux) trying to be replaced while in use.

**Comment by jankatins** [comment]
Maybe it would be interesting to replace update.packages with a version which checks if one of the to-be-updated packages are in a "blacklist" and if so errors instead of crashing...?

**Comment by takluyver** [comment]
I feel like we don't understand the problem well enough, and I'm wary of attempting to paper over bugs in R package installation  - that's the kind of fix that can end up causing as much aggravation as the problem it aims to resolve.

**Comment by jankatins** [comment]
true...

BTW: I looked into the Rsession process of rstudio and they use rcp messages to communicate between the UI and the session... Not sure how easy it would be to translate the jupyter messages to rstudio messages and then use their rsession as the backend for the kernel... would remove a lot of headache from our usage of `evaluate`...

**Comment by flying-sheep** [comment]
well, i guess that would be a separate project then.

i think a hard dependency on RStudio is out of scope for this

**Comment by jankatins** [comment]
Yep, the last would a replacement for our current evaluate based implementation...

**Comment by flying-sheep** [comment]
i’ve added most FAQ entries, but it’s not clear if we should really add this
